macros/add_property_labels.sql

```diff
@@ -0,0 +1,42 @@
+{% macro add_property_labels(passthrough_var_name, cte_name) %}
+
+select {{ cte_name }}.*
+
+{%- if var(passthrough_var_name, []) != [] -%}
+  {%- set col_list = var(passthrough_var_name) -%}
+
+  {%- for col in col_list -%} -- Create label cols
+    {%- if col.add_property_label or var('hubspot__enable_all_property_labels', false) -%}
+      {%- set col_alias = (col.alias | default(col.name)) %}
+      , {{ col.name }}_option.property_option_label as {{ col_alias }}_label
+    {% endif -%}
+  {%- endfor %}
+
+  from {{ cte_name }}
+
+  {% for col in col_list -%} -- Create joins
+    {%- if col.add_property_label or var('hubspot__enable_all_property_labels', false) -%}
+      {%- set col_alias = (col.alias | default(col.name)) %}
+
+  left join -- create subset of property and property_options for property in question
+    (select 
+      property_option.property_option_value, 
+      property_option.property_option_label
+    from {{ ref('stg_hubspot__property_option') }} as property_option
+    join {{ ref('stg_hubspot__property') }} as property
+      on property_option.property_id = property._fivetran_id
+      and property_option.property_option_label = property.property_label
+    where property.property_name = '{{ col.name.replace('property_', '') }}'
+    ) as {{ col.name }}_option
+
+    on cast({{ cte_name }}.{{ col_alias }} as {{ dbt.type_string() }})
+      = cast({{ col.name }}_option.property_option_value as {{ dbt.type_string() }})
```

Comment:
We are doing a lot of join logic (along with a subquery) here. I could see this becoming very computationally expensive if a customer utilizes this for all their fields. 

I feel it is worth calling out somewhere that this may increase the runtime of the models.

Reply:
Design wise I cannot think of a more performant approach. For the customers that want this sort of information, this is the best way to do it in my opinion. Although I do appreciate this being disabled by default and an opt in feature for those that truly want it.

Reply:
Yes, this was one of the reasons why I opted to have it work only on defined fields. I will add a note to the readme. I had a prior version where I did allow both passthrough all columns and passthrough all labels on, but removed it. There were a few reasons:
1. This took a very, very long time to run on I think, for example `company`, and the end results weren't great either. Not all hs_property fields have useful labels, so the result was most columns had null labels. I think if a user needs to bring in labels, it will only be useful for certain columsn.
2. It was a bit of a nightmare for the staging models that use some of the `hs_cols` as defacto fields, tho this is a lesser issue than above.